austation/code/game/objects/structures/party_palloons.dm

```diff
@@ -0,0 +1,84 @@
+/////////////////////  CONSUMABLE BALLOON  /////////////////////
+
+
+/obj/item/balloon
+	name = "deflated balloon"
+	desc = "A small, deflated balloon.  You can use it in your hand to inflate it."
+	icon = 'austation/icons/obj/structures/balloons.dmi'
+	icon_state = "red"
+	w_class = WEIGHT_CLASS_TINY
+	var/coloring = "red"
+	var/inflate_time = 20
+
+/obj/item/balloon/Initialize()
+	. = ..()
+	var/list/colorslist = list(
+		"red",
+		"blue",
+		"green",
+		"yellow",
+		"purple",
+		"pink"
+	)
+	color = pick(colorslist)
+	icon_state = coloring
+
+/obj/item/balloon/attack_self(mob/living/carbon/user)
+	. = ..()
+	var/turf/T = get_turf(user)
+	if(!T)
+		user.show_message("<span class='warning'> There is no room to inflate the baloon.</span>")
+		return
+	user.visible_message("<span class='notice'> [user] begins to inflate the balloon.</span>", "<span class='notice'> You begin to inflate the balloon.</span>", "", "<span class='notice'> You can hear someone blowing up a balloon.</span>", 7)
+	if(!do_mob(user, user, inflate_time))
+		return
+	if(!ishuman(user))
+		return
+	var/mob/living/carbon/human/U = user
+	if(HAS_TRAIT(U, TRAIT_NOBREATH) || !U.getorganslot(ORGAN_SLOT_LUNGS))
+		user.show_message("<span class='warning'> You can not blow up the balloon because you do not breathe!</span>")
+		return
+	U.adjustOxyLoss(20)
+	var/obj/structure/balloon/B = new(T)
+	B.coloring = coloring + "_i"
```

Comment:
coloring is never set to anything except red, on line 23 you set the color variable, not coloring.
also, the Initialize() proc has already been called on the balloon, so it's color won't be set by this.

Reply:
ah, my mistake.
it turns out that color is already a var, so I had to go around renaming all my vars to something else (coloring)

Reply:
changed